File: back/src/main/java/com/back/domain/concert/event/SeatOccupiedEvent.java
package com.back.domain.concert.event;

/**
 * 좌석 선점 완료 이벤트.
 *
 * <p>DB 트랜잭션 커밋 후 {@code @TransactionalEventListener(phase = AFTER_COMMIT)}에서
 * Redisson Delayed Queue에 만료 복구 메시지를 등록할 때 사용됩니다.
 *
 * @param concertId   공연 ID
 * @param scheduleId  회차 ID
 * @param seatNumber  좌석 번호
 * @param ttlSeconds  선점 유지 시간 (초)
 */
public record SeatOccupiedEvent(
        Long concertId,
        Long scheduleId,
        String seatNumber,
        long ttlSeconds
) {
}

File: back/src/main/kotlin/com/back/domain/concert/event/SeatOccupiedEvent.kt
package com.back.domain.concert.event

import com.back.domain.schedule.constant.SeatStatus

data class SeatOccupiedEvent(
    override val concertId: Long,
    override val scheduleId: Long,
    override val seatNumber: String,
    val ttlSeconds: Long
) : SeatEvent {
    override val status: SeatStatus get() = SeatStatus.HOLD
}
